spec/controllers/bulk_submission_excel/downloads_controller_spec.rb

```diff
@@ -11,23 +13,81 @@
     end
   end
 
-  context 'when receiving a create request' do
-    let(:submission) { create :submission }
-    let(:plates) { create_list(:plate, 2) }
+  describe 'GET #new' do
+    it 'renders the new template' do
+      get :new, params: { submission_template_id: submission.id }
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe 'POST #create' do
+    let(:plates) { create_list(:plate, 6) }
     let(:barcodes) { plates.map(&:barcodes).flatten.map(&:barcode) }
-    let(:action) do
-      post :create,
-           params: {
-             bulk_submission_excel_download: {
-               asset_barcodes: barcodes,
-               submission_template_id: submission.id
-             }
-           }
+    let(:valid_attributes) do
+      {
+        bulk_submission_excel_download: {
+          asset_barcodes: barcodes.join("\n"),
+          submission_template_id: submission.id
+        }
+      }
+    end
+    let(:invalid_attributes) do
+      { bulk_submission_excel_download: { asset_barcodes: 'invalid barcode', submission_template_id: submission.id } }
+    end
+
+    context 'with valid params' do
+      it 'creates a new download and sends a file' do
+        post :create, params: valid_attributes
+        expect(
+          response.header['Content-Type']
+        ).to include 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
```

Comment:
Can this string be a constant on subject's class so it is not repeated as string?